build-system/tasks/helpers.js

```diff
@@ -130,6 +147,46 @@ async function compileBentoRuntime(options) {
   await doBuildJs(jsBundles, 'bento.js', options);
 }
 
+/**
+ * @typedef {{
+ *  minifiedName?: string;
+ *  toName?: string;
+ *  outputFormat?: string;
+ *  esbuild?: boolean;
+ *  minify?: boolean;
+ *  watch?: boolean;
+ *  onWatchBuild?: *;
+ *  wrapper?: string;
+ *  babelCaller?: string;
+ *  remapDependencies?: Object;
+ *  externalDependencies?: Array<string>
+ * }} CompileBentoCoreOptions
+ */
+
+/**
+ * @param {CompileBentoCoreOptions} options
+ * @return {Promise<void>}
+ */
+async function compileBentoCore(options) {
+  const {options: bundleOpts, srcDir, srcFilename} = jsBundles['bento.core.js'];
+  const {minifiedName, toName} = bundleOpts;
+  const filename = `${srcDir}/${srcFilename}`;
+  const fileSource = generateBentoCoreEntrypoint();
+  await fs.outputFile(filename, fileSource);
+
+  const esm = argv.esm || argv.sxg || false;
+  await doBuildJs(jsBundles, 'bento.core.js', {
+    ...options,
+    toName: maybeToNpmEsmName(toName),
+    minifiedName: maybeToNpmEsmName(minifiedName),
+
+    outputFormat: esm ? 'esm' : 'cjs',
```

Comment:
Do we ever output `cjs` during NPM builds?

Reply:
I believe we do for the npm extension binaries [here](https://github.com/ampproject/amphtml/blob/main/build-system/tasks/extension-helpers.js#L658). I was unclear on why but I kept consistent with the logic there